module#2/src/app.ts

```diff
@@ -1,19 +1,39 @@
 import express from "express";
-
-import { routerUsers, routerCommon, routerGroups } from "./api/routes";
+import cors from "cors";
+import { routerUsers, routerCommon, routerGroups, routerAuth } from "./api/routes";
 import { authenticateMessage, PORT, PORT_FOR_DB } from "./constants";
 import { sequelize } from "./data-access";
 import { Group, User, UserGroup } from "./models";
+import { apiLogger, methodsError, morganMiddleware, routersHandleErrors } from "./middlewares";
+import { logger } from "./loaders";
+import { checkTokenAccess } from "./middlewares/authorization";
 
 const app = express();
 
 app.use(express.json());
 
-app.listen(PORT, () => console.log(`Server has been started on ${PORT} port`));
+app.use(cors());
+
+app.use(apiLogger);
+app.use(morganMiddleware);
 
 app.use("/", routerCommon);
-app.use("/users", routerUsers);
-app.use("/groups", routerGroups);
+app.use("/users", checkTokenAccess, routerUsers);
+app.use("/groups", checkTokenAccess, routerGroups);
+app.use("/auth", routerAuth);
+app.use(routersHandleErrors);
+app.use(methodsError);
+
+process
+  .on("uncaughtException", (error) => {
+    logger.error(`${error} : uncaught exception thrown`);
+    process.exit(1);
```

Comment:
If I remember correctly to finished the process with error. I thought to avoid memory leak, stop the Node.js event loop, and any asynchronous operations that are still in progress will be aborted.  Something like this.